Examples/Python/src/GeometryBuildingGen1.cpp

```diff
@@ -0,0 +1,155 @@
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace Acts::Python {
+void addGeometryBuildingGen1(Context &ctx) {
+  auto m = ctx.get("main");
+
+  using SurfacePtrVector = std::vector<std::shared_ptr<const Surface>>;
+
+  py::class_<Acts::Layer, std::shared_ptr<Acts::Layer>>(m, "Layer");
+
+  {
+    auto creator =
+        py::class_<Acts::LayerCreator>(m, "LayerCreator")
+            .def(py::init<Acts::LayerCreator::Config>())
+            .def("cylinderLayer",
+                 [](const Acts::LayerCreator &self, const GeometryContext &gctx,
+                    SurfacePtrVector surfaces, std::size_t binsPhi,
+                    std::size_t binsZ) {
+                   return self.cylinderLayer(gctx, surfaces, binsPhi, binsZ);
+                 })
+            .def("discLayer",
+                 [](const Acts::LayerCreator &self, const GeometryContext &gctx,
+                    SurfacePtrVector surfaces, std::size_t binsR,
+                    std::size_t binsPhi) {
+                   return self.discLayer(gctx, surfaces, binsR, binsPhi);
+                 });
+
+    auto config =
+        py::class_<LayerCreator::Config>(creator, "Config").def(py::init<>());
+    ;
+
+    ACTS_PYTHON_STRUCT_BEGIN(config, LayerCreator::Config);
+    ACTS_PYTHON_MEMBER(surfaceArrayCreator);
+    ACTS_PYTHON_MEMBER(cylinderZtolerance);
+    ACTS_PYTHON_MEMBER(cylinderPhiTolerance);
+    ACTS_PYTHON_STRUCT_END();
+  }
+
+  {
+    using Creator = Acts::SurfaceArrayCreator;
+    using Config = typename Creator::Config;
+
+    auto creator =
+        py::class_<Creator, std::shared_ptr<Creator>>(m, "SurfaceArrayCreator")
+            .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  {
+    using Base = Acts::ILayerArrayCreator;
+    using Creator = Acts::LayerArrayCreator;
+    using Config = typename Creator::Config;
+
+    py::class_<Base, std::shared_ptr<Base>>(m, "ILayerArrayCreator");
+
+    auto creator = py::class_<Creator, std::shared_ptr<Creator>, Base>(
+                       m, "LayerArrayCreator")
+                       .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  {
+    using Base = Acts::ITrackingVolumeArrayCreator;
+    using Creator = Acts::TrackingVolumeArrayCreator;
+    using Config = typename Creator::Config;
+
+    py::class_<Base, std::shared_ptr<Base>>(m, "ITrackingVolumeArrayCreator");
+
+    auto creator = py::class_<Creator, std::shared_ptr<Creator>, Base>(
+                       m, "TrackingVolumeArrayCreator")
+                       .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  py::class_<Acts::VolumeBounds, std::shared_ptr<Acts::VolumeBounds>>(
+      m, "VolumeBounds");
+
+  py::class_<Acts::CylinderVolumeBounds,
```

Comment:
**IF** we're deciding that volume bounds etc will be exposed in Python, would it be a good idea to add all of the other types as well? Or can we leave this on a case by case basis?

Reply:
Same as above, I think case-by-case is enough for the purpose of these bindings...

Reply:
Can you unify this with the `makeCylinderVolume` function that is in `Geometry.cpp`. I guess we don't need both?

Reply:
`makeCylinderVolume` was actually unused, so I removed it. Moved Binding for `CylinderVolumeBounds` to `Geometry.cpp` anyways because I think its better there.